### DISCUSSION TITLE
可以用 Goreleaser 构建压缩包和各种发行版的安装包

### DISCUSSION THREAD

**Discussion by sixg0000d** [v2fly/v2ray-core#1259] [Ideas]
1. 即使在对各种发行版打包不了解的情况下也能打包出 deb 、 rpm 甚至 apk （是 Alpine Linux 不是 Android ）包。
2. 代替自写 workflow 文件。它也可以集成在 Github Actions 内（见 https://goreleaser.com/ci/actions/ ），在 workflow 文件内容减少的情况下能做到同样的事，选项很多甚至更灵活。
3. [Goreleaser](https://github.com/goreleaser/goreleaser/) 和 v2ray-core 一样使用 MIT 协议，应该能放心使用。

**Comment by Loyalsoldier** [comment]
问题在于需要分发的平台太多，串行编译太慢，所以使用了 GitHub Actions 的多实例并行构建。

**Reply by sixg0000d** [reply]
[Build](https://goreleaser.com/cmd/goreleaser_build/) 和 [Release](https://goreleaser.com/cmd/goreleaser_release/) 任务应该是默认以 CPU 数为基准并行执行的，在 Github Actions 里应该也是如此。

~~Github 的 [cli](https://github.com/cli/cli/releases) 项目用 Goreleaser 打包了 deb 、rpm 和 tar.gz 三种包，[用时 8m46s ](https://github.com/cli/cli/runs/3410687071) ，其中 goreleaser 主要 step 用时 4m48s ，应该不算太慢吧。（实在不行也可以用多个带 goreleaser 的 workflow）~~

好像 v2ray-core 构建的平台是挺多的...这样吧，我先在 fork 里试试
